data_model/src/query/predicate/predicate_atoms/mod.rs

```diff
@@ -0,0 +1,185 @@
+            where
+                Proj: Fn($predicate_ty) -> OutputType + Copy,
+            {
+                CompoundPredicate::Atom(proj(self))
+            }
+        }
+
+        impl core::ops::Not for $predicate_ty
+        where
+            Self: AstPredicate<$predicate_ty>,
+        {
+            type Output = NotAstPredicate<Self>;
+
+            fn not(self) -> Self::Output {
+                NotAstPredicate(self)
+            }
+        }
+
+        impl<PRhs> core::ops::BitAnd<PRhs> for $predicate_ty
+        where
+            Self: AstPredicate<$predicate_ty>,
+            PRhs: AstPredicate<$predicate_ty>,
+        {
+            type Output = AndAstPredicate<Self, PRhs>;
+
+            fn bitand(self, rhs: PRhs) -> Self::Output {
+                AndAstPredicate(self, rhs)
+            }
+        }
+
+        impl<PRhs> core::ops::BitOr<PRhs> for $predicate_ty
+        where
+            Self: AstPredicate<$predicate_ty>,
+            PRhs: AstPredicate<$predicate_ty>,
+        {
+            type Output = OrAstPredicate<Self, PRhs>;
+
+            fn bitor(self, rhs: PRhs) -> Self::Output {
+                OrAstPredicate(self, rhs)
+            }
+        }
+    };
+}
+pub(crate) use impl_predicate_box;
+
+/// A predicate that can be applied to a [`String`]-like types.
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+pub enum StringPredicateBox {
+    /// Checks if the input is equal to the expected value
+    Equals(String),
+    /// Checks if the input contains an expected substring, like [`str::contains()`]
+    Contains(String),
+    /// Checks if the input starts with an expected substring, like [`str::starts_with()`]
+    StartsWith(String),
+    /// Checks if the input ends with an expected substring, like [`str::ends_with()`]
+    EndsWith(String),
+}
+
+impl_predicate_box!(String, Name: StringPredicateBox);
+
+impl<T> PredicateTrait<T> for StringPredicateBox
+where
+    T: AsRef<str>,
+{
+    fn applies(&self, input: &T) -> bool {
+        let input = input.as_ref();
+        match self {
+            StringPredicateBox::Contains(content) => input.contains(content),
+            StringPredicateBox::StartsWith(content) => input.starts_with(content),
+            StringPredicateBox::EndsWith(content) => input.ends_with(content),
+            StringPredicateBox::Equals(content) => *input == *content,
+        }
+    }
+}
+
+/// A predicate that can be applied to [`Metadata`].
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+pub enum MetadataPredicateBox {
+    // TODO: populate this with something. Seeing as how we can change it to be just a JsonString, not populating it right now
+}
+
+impl_predicate_box!(Metadata: MetadataPredicateBox);
+
+impl PredicateTrait<Metadata> for MetadataPredicateBox {
+    fn applies(&self, _input: &Metadata) -> bool {
+        match *self {}
```

Comment:
Thought for some time how this result in `bool`...
Before realizing that `!` can be substitute any type.

Reply:
me too